htdocs/asset/document.php

```diff
@@ -56,100 +59,79 @@
 if (!$sortorder) $sortorder = "ASC";
 if (!$sortfield) $sortfield = "name";
 
-// Initialize technical objects
 $object = new Asset($db);
-$extrafields = new ExtraFields($db);
-$diroutputmassaction = $conf->assets->dir_output.'/temp/massgeneration/'.$user->id;
-$hookmanager->initHooks(array('assetdocument')); // Note that conf->hooks_modules contains array
-
-// Fetch optionals attributes and labels
-$extrafields->fetch_name_optionals_label($object->table_element);
-
-// Load object
-include DOL_DOCUMENT_ROOT.'/core/actions_fetchobject.inc.php'; // Must be include, not include_once  // Must be include, not include_once. Include fetch and fetch_thirdparty but not fetch_optionals
-
-// Security check
-if (!empty($user->socid)) $socid = $user->socid;
-$result = restrictedArea($user, 'asset', $id);
-
-//if ($id > 0 || ! empty($ref)) $upload_dir = $conf->sellyoursaas->multidir_output[$object->entity] . "/packages/" . dol_sanitizeFileName($object->id);
-if ($id > 0 || !empty($ref)) $upload_dir = $conf->sellyoursaas->multidir_output[$object->entity]."/packages/".dol_sanitizeFileName($object->ref);
-
+if ($object->fetch($id))
+{
+	$upload_dir = $conf->asset->dir_output."/".dol_sanitizeFileName($object->ref);
+}
 
 
 /*
  * Actions
  */
 
-include_once DOL_DOCUMENT_ROOT.'/core/actions_linkedfiles.inc.php';
+require_once DOL_DOCUMENT_ROOT.'/core/actions_linkedfiles.inc.php';
```

Comment:
@aspangaro I think it's include here, not require_once or include_once

Reply:
Hi @frederic34

Hum, I found require_once or include_once everywhere but no include.
It would potentially need to be harmonized. Why not use require_once? The file being necessary for this part to work properly...